src/DynamoCoreWpf/ViewModels/PackageManager/PackageManagerSearchElementViewModel.cs

```diff
@@ -58,24 +73,55 @@ private static void GoToUrl(string url)
             }
         }
 
-        public List<Tuple<PackageVersion, DelegateCommand>> Versions
+        public List<Tuple<PackageVersion, DelegateCommand<object>>> Versions
         {
             get
             {
                 return
                     Model.Header.versions.Select(
-                        x => new Tuple<PackageVersion, DelegateCommand>(x, new DelegateCommand(() => OnRequestDownload(x)))).Reverse().ToList();
+                        x => new Tuple<PackageVersion, DelegateCommand<object>>(
+                            x, new DelegateCommand<object>((p) => OnRequestDownload(x, p.Equals("true")))
+                        )).Reverse().ToList();
             }
         }
 
-        public delegate void PackageSearchElementDownloadHandler(PackageManagerSearchElement element, PackageVersion version);
+        private List<String> CustomPackageFolders;
+
+        public delegate void PackageSearchElementDownloadHandler(
+            PackageManagerSearchElement element, PackageVersion version, string downloadPath = null);
         public event PackageSearchElementDownloadHandler RequestDownload;
-        public void OnRequestDownload(PackageVersion version)
+
+        public void OnRequestDownload(PackageVersion version, bool downloadToCustomPath)
         {
-            if (RequestDownload != null)
+            string downloadPath = null;
+
+            if (downloadToCustomPath)
             {
-                RequestDownload(this.Model, version);
+                downloadPath = GetDownloadPath();
+
+                if (downloadPath == null)
+                    return;
             }
+
+            if (RequestDownload != null)
+                RequestDownload(this.Model, version, downloadPath);
+        }
+
+        private string GetDownloadPath()
+        {
+            var args = new PackagePathEventArgs();
+
+            ShowFileDialog(args);
+
+            if (args.Cancel)
+                return null;
```

Comment:
It's generally a good idea for a method to return `String.Empty` (if an API returns an array or a collection, it's better to return an empty collection rather than a `null` value).